### PR TITLE
[BEAM-14503] Add support for Flink 1.15

### PR DESCRIPTION
**Pull request by jto**
Flink 1.15 has a number of improvements. Most notably it completely shades its dependency on Scala so Beam does not need to pick a specific version to support anymore.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by jto** [comment]
@ibzib Since you implemented support for 1.14, can you please take a look ?

File: runners/flink/1.12/build.gradle [hunk 1]
```diff
@@ -22,8 +22,8 @@ project.ext {
   // Set the version of all Flink-related dependencies here.
   flink_version = '1.12.7'
   // Version specific code overrides.
-  main_source_overrides = ['./src/main/java']
-  test_source_overrides = ['./src/test/java']
+  main_source_overrides = ["${basePath}/1.11/src/main/java", './src/main/java']
```
Comment (ibzib) [inline runners/flink/1.12/build.gradle:25]: Flink 1.11 has been deleted from Beam. 1.12 is the current minimum version.

I realized I accidentally left a file in the Flink 1.11 directory, sent a PR to fix: #17750

**Review by ibzib — COMMENTED** [review]
Thanks for contributing!

Make sure to request a beam_flink1.15_job_server be created on Docker hub (like https://issues.apache.org/jira/browse/INFRA-22825).

File: runners/flink/1.12/build.gradle (hunk 1 above)
Comment (jto) [reply runners/flink/1.12/build.gradle]: Ah yes I did not realize that :) Thanks for the quick response. I fixed the PR accordingly.

**Review by ibzib — APPROVED** [review]
Looks good!

FYI @kileys - we will need to add a new Docker repo to release this.

**Comment by kileys** [comment]
Thanks Kyle! Filed https://issues.apache.org/jira/browse/INFRA-23301

**Pull request merged**